launcher/tasks/ConcurrentTask.h

```diff
@@ -41,14 +41,17 @@
 #include <QUuid>
 #include <memory>
 
+#include "Application.h"
 #include "tasks/Task.h"
 
 class ConcurrentTask : public Task {
     Q_OBJECT
    public:
     using Ptr = shared_qobject_ptr<ConcurrentTask>;
 
-    explicit ConcurrentTask(QObject* parent = nullptr, QString task_name = "", int max_concurrent = 6);
+    explicit ConcurrentTask(QObject* parent = nullptr,
+                            QString task_name = "",
+                            int max_concurrent = APPLICATION->settings()->get("NumberOfConcurrentTasks").toInt());
```

Comment:
This is a bad Idea. Placing a call to `APPLICATION` in a header restricts the header's use to the main exe.
The call should be made and passed to the class where it's needed but eh class default should still be a fixed value

Reply:
Yeah I had a similar issue with the test, decided to go with a #ifdef in the two cases the tests were failing

Reply:
resolved?

Reply:
Partially: I still use applications but in the cpp files. 
The reason I kept this open is that I added some code(a define in cmake) to determine if I should use the setting or not based if we run tests or not.
Also wanted to get Ryes opinion on the current implementation.